From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stoiko Ivanov <s.ivanov@proxmox.com>
Date: Thu, 23 May 2019 15:32:53 +0200
Subject: [PATCH] tests: fix cosmetic permission issues during `make install`

files in dist_*_SCRIPTS get installed with 0755, those in dist_*_DATA
with 0644. This commit moves all .kshlib, .shlib and .cfg files in the
testsuite to dist_pkgdata_DATA, and removes the shebang from
zpool_import.kshlib.

This ensures that the files are installed with appropriate permissions
and silences some warnings from lintian

Signed-off-by: Stoiko Ivanov <s.ivanov@proxmox.com>
---
 tests/zfs-tests/tests/functional/alloc_class/Makefile.am            | 6 ++++--
 .../tests/functional/cli_root/zpool_import/zpool_import.kshlib      | 2 --
 .../tests/functional/cli_root/zpool_initialize/Makefile.am          | 1 -
 tests/zfs-tests/tests/functional/cli_root/zpool_trim/Makefile.am    | 4 +++-
 tests/zfs-tests/tests/functional/removal/Makefile.am                | 5 ++++-
 tests/zfs-tests/tests/perf/Makefile.am                              | 2 +-
 6 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/tests/zfs-tests/tests/functional/alloc_class/Makefile.am b/tests/zfs-tests/tests/functional/alloc_class/Makefile.am
index 073eac988..7cffb2eac 100644
--- a/tests/zfs-tests/tests/functional/alloc_class/Makefile.am
+++ b/tests/zfs-tests/tests/functional/alloc_class/Makefile.am
@@ -1,7 +1,5 @@
 pkgdatadir = $(datadir)/@PACKAGE@/zfs-tests/tests/functional/alloc_class
 dist_pkgdata_SCRIPTS = \
-	alloc_class.cfg \
-	alloc_class.kshlib \
 	setup.ksh \
 	cleanup.ksh \
 	alloc_class_001_pos.ksh \
@@ -17,3 +15,7 @@ dist_pkgdata_SCRIPTS = \
 	alloc_class_011_neg.ksh \
 	alloc_class_012_pos.ksh \
 	alloc_class_013_pos.ksh
+
+dist_pkgdata_DATA = \
+	alloc_class.cfg \
+	alloc_class.kshlib
diff --git a/tests/zfs-tests/tests/functional/cli_root/zpool_import/zpool_import.kshlib b/tests/zfs-tests/tests/functional/cli_root/zpool_import/zpool_import.kshlib
index f53b88f79..d050145e4 100644
--- a/tests/zfs-tests/tests/functional/cli_root/zpool_import/zpool_import.kshlib
+++ b/tests/zfs-tests/tests/functional/cli_root/zpool_import/zpool_import.kshlib
@@ -1,5 +1,3 @@
-#!/bin/ksh
-
 #
 # This file and its contents are supplied under the terms of the
 # Common Development and Distribution License ("CDDL"), version 1.0.
diff --git a/tests/zfs-tests/tests/functional/cli_root/zpool_initialize/Makefile.am b/tests/zfs-tests/tests/functional/cli_root/zpool_initialize/Makefile.am
index a0a0e0b5c..2ebc376d9 100644
--- a/tests/zfs-tests/tests/functional/cli_root/zpool_initialize/Makefile.am
+++ b/tests/zfs-tests/tests/functional/cli_root/zpool_initialize/Makefile.am
@@ -3,7 +3,6 @@ dist_pkgdata_SCRIPTS = \
 	cleanup.ksh \
 	zpool_initialize_attach_detach_add_remove.ksh \
 	zpool_initialize_import_export.ksh \
-	zpool_initialize.kshlib \
 	zpool_initialize_offline_export_import_online.ksh \
 	zpool_initialize_online_offline.ksh \
 	zpool_initialize_split.ksh \
diff --git a/tests/zfs-tests/tests/functional/cli_root/zpool_trim/Makefile.am b/tests/zfs-tests/tests/functional/cli_root/zpool_trim/Makefile.am
index c357eeffb..d2d3b4ae8 100644
--- a/tests/zfs-tests/tests/functional/cli_root/zpool_trim/Makefile.am
+++ b/tests/zfs-tests/tests/functional/cli_root/zpool_trim/Makefile.am
@@ -2,7 +2,6 @@ pkgdatadir = $(datadir)/@PACKAGE@/zfs-tests/tests/functional/cli_root/zpool_trim
 dist_pkgdata_SCRIPTS = \
 	setup.ksh \
 	cleanup.ksh \
-	zpool_trim.kshlib \
 	zpool_trim_attach_detach_add_remove.ksh \
 	zpool_trim_import_export.ksh \
 	zpool_trim_multiple.ksh \
@@ -20,3 +19,6 @@ dist_pkgdata_SCRIPTS = \
 	zpool_trim_unsupported_vdevs.ksh \
 	zpool_trim_verify_checksums.ksh \
 	zpool_trim_verify_trimmed.ksh
+
+dist_pkgdata_DATA = \
+	zpool_trim.kshlib
diff --git a/tests/zfs-tests/tests/functional/removal/Makefile.am b/tests/zfs-tests/tests/functional/removal/Makefile.am
index c5d013e7c..ba42b899a 100644
--- a/tests/zfs-tests/tests/functional/removal/Makefile.am
+++ b/tests/zfs-tests/tests/functional/removal/Makefile.am
@@ -28,6 +28,9 @@ dist_pkgdata_SCRIPTS = \
 	removal_with_send.ksh removal_with_send_recv.ksh \
 	removal_with_snapshot.ksh removal_with_write.ksh \
 	removal_with_zdb.ksh remove_mirror.ksh remove_mirror_sanity.ksh \
-	remove_raidz.ksh remove_expanded.ksh removal.kshlib
+	remove_raidz.ksh remove_expanded.ksh
+
+dist_pkgdata_DATA = \
+	removal.kshlib
 
 pkgexecdir = $(datadir)/@PACKAGE@/zfs-tests/tests/functional/removal
diff --git a/tests/zfs-tests/tests/perf/Makefile.am b/tests/zfs-tests/tests/perf/Makefile.am
index 68dd31ec1..294b136b3 100644
--- a/tests/zfs-tests/tests/perf/Makefile.am
+++ b/tests/zfs-tests/tests/perf/Makefile.am
@@ -1,5 +1,5 @@
 pkgdatadir = $(datadir)/@PACKAGE@/zfs-tests/tests/perf
-dist_pkgdata_SCRIPTS = \
+dist_pkgdata_DATA = \
 	nfs-sample.cfg \
 	perf.shlib
 
